core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -292,13 +298,16 @@ private[spark] class TaskSchedulerImpl(
    * given TaskSetManager have completed, so state associated with the TaskSetManager should be
    * cleaned up.
    */
-  def taskSetFinished(manager: TaskSetManager): Unit = synchronized {
+  def taskSetFinished(manager: TaskSetManager, success: Boolean): Unit = synchronized {
```

Comment:
Can't we clean up the `stageIdToFinishedPartitions` when there is no more task set manager for the stage? Then we can avoid adding a new parameter.

Reply:
When a zombie TaskSetManager finished, it could also lead to empty TaskSetManager for the stage(just see `taskSetFinished()`).  So, if we clean up the `stageIdToFinishedPartitions` when there is no more task set manager for the stage, the active TaskSetManager which is created later would lost finished partitions. WDYT ? @cloud-fan

Reply:
Possibly, we could move the clean up logic into `TaskSetManager.maybeFinishTaskSet()`, then we could avoid adding a new parameter here. But it seems a little weird to clean a TaskSchedulerImpl's stage related variable in TSM rather than TaskSchedulerImpl. WDYT? @cloud-fan